File: packages/zynth-markdown/android/ZynthMarkdown/src/main/java/dev/zynth/markdown/ZynthMarkdown.kt
package dev.zynth.markdown

import android.app.Activity
import android.util.Log
import com.zynth.kit.runtime.ZynthRuntime

/**
 * Public interface for ZynthMarkdown module.
 */
object ZynthMarkdown {
    private var moduleInstance: ZynthMarkdownModule? = null

    @JvmStatic
    fun initialize(activity: Activity, runtime: ZynthRuntime) {
        if (moduleInstance != null) {
            Log.w("ZynthMarkdown", "Module already initialized")
            return
        }

        moduleInstance = ZynthMarkdownModule()
        runtime.installModules(listOf(moduleInstance!!))
        Log.d("ZynthMarkdown", "Module initialized")
    }

    @JvmStatic
    fun cleanup() {
        moduleInstance = null
    }
}

File: packages/zynth-markdown/ios/ZynthMarkdown.swift
import Foundation
import ZynthKit

@objc(ZynthMarkdown)
public class ZynthMarkdown: NSObject {
  private static var moduleInstance: ZynthMarkdownModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard moduleInstance == nil else {
      print("[ZynthMarkdown] Module already initialized")
      return
    }

    let module = ZynthMarkdownModule()
    runtime.installModules([module])
    moduleInstance = module
    print("[ZynthMarkdown] Module initialized")
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}
